### PR TITLE
File field formatters port

### PR DESCRIPTION
**Pull request by chindris**

### CODE REVIEW

**Comment by chindris** [comment]
I did a rebase instead of a merge and I had to delete the previous branch used for the pull request here: https://github.com/drupal-media/file_entity/pull/2 . All the issues from the previous pull request should be fixed, this is ready for a new review.

**Comment by Berdir** [comment]
Note: You can easily do force push for pull request branches, that works just fine :)

**Comment by chindris** [comment]
oh, yes, forgot that you can do that...

**Comment by Berdir** [comment]
This looks pretty good to me.

I think file_entity_field_formatter_info_alter() is the only thing remaining in field.inc, right? I'd prefer to get rid of those .inc files and hardcoded inclusion of them in .module. so maybe move that to .module? It's currently completely disabled with a return. The second part can definitely be removed, not sure about those descriptions, are they worth keeping?

I think we should also try to make the entity_reference_entity_view formatter available for file and image field types, as a replacement for the render file one that we are removing here.

The last point is tests, all this is currently completely untested. Not sure what to do about that. I'm open to merging it in without them but it would be great to have some tests if you have time for writing them. We could also merge this now and work on tests in a separate PR.

**Comment by slashrsm** [comment]
Yes, let's fix the alter that @Berdir and do tests in a follow-up.

**Comment by chindris** [comment]
Ok, will work on this today (most probably in the evening) so that we can get this committed. Then, I can work on the tests in a follow-up.

**Comment by chindris** [comment]
I moved the file_entity_field_formatter_info_alter() into the module and made the changes @Berdir suggested. Ready for review.

File: file_entity.module [hunk 1]
```diff
@@ -126,6 +126,30 @@ function file_entity_action_info_alter(&$actions) {
...
+function file_entity_field_formatter_info_alter(array &$info) {
+  // Make the entity reference view formatter available for files and images.
+  if (!empty($info['entity_reference_entity_view'])) {
+    $info['entity_reference_entity_view']['field_types'][] = 'file';
+    $info['entity_reference_entity_view']['field_types'][] = 'image';
+  }
+
+  // Add descriptions to core formatters.
+  $descriptions = array(
+    'file_default' => t('Create a simple link to the file. The link is prefixed by a file type icon and the name of the file is used as the link text'),
```
Comment (slashrsm) [inline file_entity.module:140]: Let's finish this sentence with a period.

**Comment by slashrsm** [comment]
Merged with a rebase on top of current HEAD. Fixed that dot in the description while doing that. 

Thank you! This is great stuff!

**Comment by chindris** [comment]
Thx! @slashrsm let me know when you do a card for writing tests.

**Comment by slashrsm** [comment]
Here you go
https://trello.com/c/bSfpmKaP/86-fe-tests-write-test-coverage-for-formatters

**Pull request closed without merging**